File: frontend/src/types/address.ts
export interface Address {
  address_id: string;
  full_name: string;
  phone_country_code: string;
  phone: string;
  street: string;
  no_number: boolean;
  province: string;
  locality: string;
  zip: string | null;
  zip_unknown: boolean;
  department: string | null;
  is_default: boolean;
  created_at: string;
}

export interface AddressInput {
  full_name: string;
  phone_country_code?: string;
  phone: string;
  street: string;
  no_number?: boolean;
  province: string;
  locality: string;
  zip?: string | null;
  zip_unknown?: boolean;
  department?: string | null;
  is_default?: boolean;
}

// Las 24 provincias argentinas (23 provincias + CABA).
export const ARGENTINE_PROVINCES = [
  "Buenos Aires",
  "Ciudad Autónoma de Buenos Aires",
  "Catamarca",
  "Chaco",
  "Chubut",
  "Córdoba",
  "Corrientes",
  "Entre Ríos",
  "Formosa",
  "Jujuy",
  "La Pampa",
  "La Rioja",
  "Mendoza",
  "Misiones",
  "Neuquén",
  "Río Negro",
  "Salta",
  "San Juan",
  "San Luis",
  "Santa Cruz",
  "Santa Fe",
  "Santiago del Estero",
  "Tierra del Fuego",
  "Tucumán",
] as const;

File: backend/app/schemas/address.py
from datetime import datetime

from pydantic import BaseModel


class AddressCreate(BaseModel):
    full_name: str
    phone_country_code: str = "+54"
    phone: str
    street: str
    no_number: bool = False
    province: str
    locality: str
    zip: str | None = None
    zip_unknown: bool = False
    department: str | None = None
    is_default: bool = False


class AddressUpdate(BaseModel):
    full_name: str | None = None
    phone_country_code: str | None = None
    phone: str | None = None
    street: str | None = None
    no_number: bool | None = None
    province: str | None = None
    locality: str | None = None
    zip: str | None = None
    zip_unknown: bool | None = None
    department: str | None = None
    is_default: bool | None = None


class AddressResponse(BaseModel):
    address_id: str
    full_name: str
    phone_country_code: str
    phone: str
    street: str
    no_number: bool
    province: str
    locality: str
    zip: str | None = None
    zip_unknown: bool
    department: str | None = None
    is_default: bool
    created_at: datetime
